### PR TITLE
add --check flag to cargo fmt

### LINKED ISSUE #3888: Add `--check` command-line flag to `cargo-fmt`
cc @calebcartwright

### PR DESCRIPTION
**Pull request by calebcartwright**
Resolves #3888

### CODE REVIEW

File: src/cargo-fmt/main.rs [hunk 1]
```diff
@@ -104,6 +108,12 @@ fn execute() -> i32 {
 
     let strategy = CargoFmtStrategy::from_opts(&opts);
     let mut rustfmt_args = opts.rustfmt_options;
+    if opts.check {
+        let check_flag = String::from("--check");
```
Comment (topecongiro) [inline src/cargo-fmt/main.rs]: Nit: we can defer the allocation of a `String` until we absolutely need it:

```rust
let check_flag = "--check";
if !rustfmt_args.contains(check_flag) {
    rustfmt_args.push(check_flag.to_owned());
}
```

Comment (calebcartwright) [reply src/cargo-fmt/main.rs]: Unfortunately I believe the limitations with Vec::contains that we ran into in https://github.com/rust-lang/rustfmt/pull/3752#discussion_r317426942 still exist which prevent using a literal as an arg. 

On a related note, I was going to volunteer to work on https://github.com/rust-lang/rustfmt/issues/3587 and while working on that I'll also look into consolidating the way `cargo fmt` is transforming/passing flags to `rustfmt` which I think may help remove smoe of these Vec::contains

Comment (topecongiro) [reply src/cargo-fmt/main.rs]: Oh my god, I am so dumb :weary: Sorry about that!

Comment (calebcartwright) [reply src/cargo-fmt/main.rs]: No worries!

**Pull request merged**